simde/arm/neon/dot.h

```diff
@@ -164,6 +165,60 @@ simde_vdotq_u32(simde_uint32x4_t r, simde_uint8x16_t a, simde_uint8x16_t b) {
   #define vdotq_u32(r, a, b) simde_vdotq_u32((r), (a), (b))
 #endif
 
+#if defined(SIMDE_ARM_NEON_BFLOAT16)
```

Comment:
Huh, what's the point then? Wouldn't any Arm NEON CPU with bf16 support also have native support for these intrinsics?

In fact, `SIMDE_ARM_NEON_BFLOAT16` is never defined in this PR, so it is dead code :-(

Reply:
The initial idea is to pre-implement `bf16`-related intrinsics, and they can be quickly used when SIMDe supports `bf16` in the future.

I have now added the `bf16` types if the machine is supported. Since the `bf16` type is an option supported after `armv8.2-a`, it should add `+bf16`. Consequently, I modified two crossfiles: `aarch64-gcc-12-ccache.cross` and `aarch64-clang-15-ccache.cross` to test the `bf16`-related intrinsic.
Currently, only `vld1_bf16_x4` has been tested. The other functions are not yet completed, but I will finish the remaining part as soon as possible.

If possible, could we skip the review of other `bf16`-related functions for now and focus on reviewing other parts? Thank you for your attention, and any recommendations are welcome!

Reply:
Thanks for the `bf16` additions!

1. The point of SIMDe is to have portable implementations
2. The best way to get me to not review some code is to not have it in a PR :-) 

So lets move the bf16 code to a follow-on PR, and then I can focus on the portable code in the rest of this PR

Reply:
Okay, I will remove the `bf16` related code here and send another PR in the future.